src/patternfly/components/NotificationBadge/notification-badge.scss

```diff
@@ -1,35 +1,63 @@
+
+  // ::after border treatment
+  --pf-c-notification-badge--after--BorderColor: var(--pf-global--BackgroundColor--dark-200);
+  --pf-c-notification-badge--after--BorderRadius: var(--pf-global--BorderRadius--md);
   --pf-c-notification-badge--after--BorderWidth: var(--pf-global--BorderWidth--sm);
+  --pf-c-notification-badge--after--ZIndex: var(--pf-global--ZIndex--xs);
   --pf-c-notification-badge--after--Top: 0;
   --pf-c-notification-badge--after--Right: 0;
-  --pf-c-notification-badge--after--Width: calc(var(--pf-global--spacer--sm) + var(--pf-c-notification-badge--after--BorderWidth) + var(--pf-c-notification-badge--after--BorderWidth));
-  --pf-c-notification-badge--after--Height: calc(var(--pf-global--spacer--sm) + var(--pf-c-notification-badge--after--BorderWidth) + var(--pf-c-notification-badge--after--BorderWidth));
+  --pf-c-notification-badge--after--Bottom: 0;
+  --pf-c-notification-badge--after--Left: 0;
+  --pf-c-notification-badge--after--Width: auto;
+  --pf-c-notification-badge--after--Height: auto;
   --pf-c-notification-badge--after--BackgroundColor: transparent;
 
   // This translation works for the Bell icon, although adjusted for a happy medium between FF and Chrome.
   // If other icons are used, a modifier may need to be created to allow for proper positioning on that icon.
-  --pf-c-notification-badge--after--TranslateX: 50%;
-  --pf-c-notification-badge--after--TranslateY: calc(var(--pf-c-notification-badge--after--BorderWidth) * -1);
+  --pf-c-notification-badge--after--TranslateX: 0;
+  --pf-c-notification-badge--after--TranslateY: 0;
 
   // The icon is normalized to be square in order to facilitate consistency in positioning
   --pf-c-notification-badge__i--Width: var(--pf-global--spacer--md);
   --pf-c-notification-badge__i--Height: var(--pf-global--spacer--md);
 
   // Modifiers
+  --pf-c-notification-badge--m-read--after--BorderColor: var(--pf-global--BackgroundColor--dark-200);
   --pf-c-notification-badge--m-read--after--BackgroundColor: transparent;
-  --pf-c-notification-badge--m-read--after--BorderColor: transparent;
-  --pf-c-notification-badge--m-unread--after--BackgroundColor: var(--pf-global--active-color--400);
+
+  // Badge count
+  --pf-c-notification-badge--m-unread--after--BorderColor: var(--pf-global--active-color--100);
+  --pf-c-notification-badge--m-unread--after--BorderWidth: var(--pf-global--BorderWidth--md);
+  --pf-c-notification-badge--m-unread--after--BackgroundColor: transparent;
+
+  // Badge count
+  --pf-c-notification-badge--m-attention--Color: var(--pf-global--Color--light-100);
+  --pf-c-notification-badge--m-attention--after--BorderColor: transparent;
+  --pf-c-notification-badge--m-attention--after--BackgroundColor: var(--pf-global--danger-color--100);
+
+  // Badge count
+  --pf-c-notification-badge__count--MarginLeft: var(--pf-global--spacer--xs);
 
   position: relative;
   display: inline-block;
```

Comment:
wdyt about changing this to `inline-flex` with `align-items`? That would also normalize the spacing here in case either our or an apps react removes the space in the HTML between the icon and the count.